### PR TITLE
add exclude samples option in sample report

### PR DESCRIPTION
**Pull request by senthil10**
as of now `--exclude_sample_ids` option works only for project report, it is quite handy for having them to work for sample reports as well.

### CODE REVIEW

**Comment by remiolsen** [comment]
Good to merge.

**Comment by senthil10** [comment]
haha,. Tacka tack :)

File: scilifelab/report/delivery_notes.py [hunk 1]
```diff
@@ -206,10 +206,32 @@ def _set_project_sample_dict(project_sample_item, source):
...
+    """Check whether we should exclude a sample id.
+
+    :param exclude_sample_ids: dictionary of sample:barcode pairs
+    :param sample_name: project sample name
+    :param barcode_seq: the barcode sequence
+
+    :returns: True if exclude, False otherwise
+    """
+    if exclude_sample_ids and sample_name in exclude_sample_ids.keys():
+        if exclude_sample_ids[sample_name]:
```
Comment (Galithil) [inline scilifelab/report/delivery_notes.py]: if "exclude_sample_ids" is None( or "", or [], or {}) the function will return None, is that okay ?
same thing if 'sample_name'  is not in exclude_sample_ids.
I'm also pretty sure you can write "sample_name in exculde_sample_ids", without adding the .keys, that should be the default behaviour of a dict. 

I'm not sure I follow the logic of this, I'll have another go later

Comment (senthil10) [inline scilifelab/report/delivery_notes.py]: I did not write this function, I am just using it :wink: . And yes, it is fine if it returns `None`. As we call the function in a `if` statement below, returning `None` will just result in not entering the `if` block which is exactly what we need :+1:

**Pull request merged**